ciphey/basemods/Decoders/tap_code.py

```diff
@@ -0,0 +1,57 @@
+# by https://github.com/RustyDucky
+
+from typing import Optional, Dict, List
+
+from ciphey.iface import Config, ParamSpec, T, U, Decoder, registry
+
+
+@registry.register
+class Tap_code(Decoder[str, str]):
+    def decode(self, input: T) -> Optional[U]:
+        try: #So it doesn't throw an error when its not the right format or just not tap code
+            medium = ' '.join(map(str, input)).split(" ") #Dealing with the format of user input
+            output = ""
+
+            xone = ["A", "B", "C/K", "D", "E"]
+            xtwo =  ["F", "G", "H", "I", "J"]
+            xthr =  ["L", "M", "N", "O", "P"]  #Making the tap code table
+            xfour = ["Q", "R", "S", "Y", "U"]
+            xfive = ["V", "W", "X", "Y", "Z"]
+
+            for combination in medium:
+                y,x = combination.split(",")
+                x = int(x)
+                y = int(y)
+
+                if x == 1:
+                    output += xone[y-1]
+
+                elif x == 2:
+                    output += xtwo[y-1]
+
+                elif x == 3:
+                    output += xthr[y-1]
+
+                elif x == 4:
+                    output += xfour[y-1]
+
+                else:
+                    output += xfive[y-1]
+            return (output)
+        except:
+            return(input)

```

Comment:
Since input is a builtin python function, you should not use it as a variable name 
I think in most decoders/crackers, the incoming text is called `ctext`, so that would be a good option :)